src/main/java/com/imgix/URLBuilder.java

```diff
@@ -14,7 +15,13 @@
     private boolean useHttps;
     private String signKey;
     private boolean includeLibraryParam;
-    private final ArrayList<Integer> SRCSET_TARGET_WIDTHS = this.targetWidths();
+
+    private static final ArrayList<Integer> SRCSET_TARGET_WIDTHS = targetWidths();
+    private static final int SRCSET_WIDTH_TOLERANCE = 8;
+    private static final int MIN_WIDTH = 100;
+    private static final int MAX_WIDTH = 8192;
+    private static final Integer[] DPR_QUALITIES = {0, 75, 50, 35, 23, 20};
```

Comment:
I'm guessing the `0` here is to offset the index later in `createSrcSetDPR`?

Reply:
You are correct! However, I did want to get your opinion on this. We are mapping `[1, .., 5]` to `[75, .., 20]` I don't mind using a hash map here, but I thought since we are mapping 5 positive integers to 5 positive integers an array could suffice? If you feel it would be more clear to use a hash map, then I'm on board with that as well.

Reply:
I don't have a strong opinion but personally the extra `0` is a little off putting for me. I think a hash map would be perfectly fine. Or even _weirder_ on L217: `srcsetParams.put("q", DPR_QUALITIES[ratio - 1].toString());`